soroban-env-host/src/cost_runner/cost_types/compute_ed25519_pubkey.rs

```diff
@@ -22,10 +22,11 @@ impl CostRunner for ComputeEd25519PubKeyRun {
     }
 
     fn run_baseline_iter(
-        _host: &crate::Host,
+        host: &crate::Host,
         _iter: u64,
         sample: Self::SampleType,
     ) -> Self::RecycledType {
+        black_box(host.charge_budget(Self::COST_TYPE, 1, None).unwrap());
```

Comment:
I feel like you could make `run_baseline_iter` an defaulted member of the trait if `Self::RecycledType : From<Self:::SampleType>`, no?

Reply:
In most cases the `RecycledType` contains the `SampleType` plus the return type of the function under measurement. So we can't require `Self::RecycledType : From<Self::SampleType>`?

Reply:
I know you've previously resolved this but I just wanted to give this suggestion (what I understood) a try. 

I tried defining the `RecycledType` as a tuple `(Option<SampleType>, Option<ReturnType>)` and to implement the `run_baseline_iter` as "charge + return the sample". This won't work for a couple of reasons:

1. Associated type is an unstable feature, without it the code in  `HostCostMeasurement` looks very ugly, since it needs to set up a container `Vec<RecycledType>` the recycled results. (This is bad but not deal breaker)
2. The `SampleType` is not always returned in full, in some cases, part of it is consumed by the runner and other part of it needs to be returned. So the general definition above doesn't work, and each runner type really needs to define its own RecycledType.

Reply:
ok, cool. thanks for exploring!